README.md

```diff
@@ -1,29 +1,41 @@
-# Caramel: A fast read-only key-value store
+# CaramelDB: A High-Performance Succinct Retrieval Library
 
-This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems, computational biology, and more.
+CaramelDB is a performant C++/Python library of succinct retrieval data structures with applications in data management, machine learning systems, computational biology, and more! The centerpiece of CaramelDB is an implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient table with state-of-the-art performance on practical succinct retrieval workloads. 
+
+## What is Succinct Retrieval?
+
+As defined in [Dillinger, et al. (2022)](https://drops.dagstuhl.de/storage/00lipics/lipics-vol233-sea2022/LIPIcs.SEA.2022.4/LIPIcs.SEA.2022.4.pdf), a *retrieval data structure* $D$ for a static function $f \colon S \to \{0, 1\}^r$ is an index that returns $f(x)$ for any $x \in S$ and any value for $x \notin S$. A highly desirable goal in both theory and practice is to represent $D$ with as little space as possible. To that end, a *succinct retrieval data structure* is an index that uses only $(1 + o(1))r|S|$ bits of space, just slightly more than the information-theoretic lower bound of $r|S|$ bits.
```

Comment:
what does the {0, 1}^r part mean?

Reply:
It is the set of binary strings of length r